woocommerce/woocommerce
# Fix lost password page visibility when logged in

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Prevents access to the lost password page when logged in. Also cleans up rendering of account page to remove deprecated elements and split logged in/logged out behaviour (which I implemented before the redirect).

Closes #28956



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. As a logged in user go to the my account page
2. Navigate between each section and ensure things look correct
3. Navigate directly to /my-account/lost-password. You should be redirected to the "edit account" page.
4. Log out.
5. On the account page click "lost password"
6. Ensure the lost password form is shown and you can go through the full account reset flow



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
